Client/mods/deathmatch/logic/CClientVehicleManager.cpp

```diff
@@ -385,60 +162,59 @@ void CClientVehicleManager::DoPulse()
 CClientVehicle* CClientVehicleManager::Get(ElementID ID)
 {
     // Grab the element with the given id. Check its type.
-    CClientEntity* pEntity = CElementIDs::GetElement(ID);
+    auto pEntity = CElementIDs::GetElement(ID);
     if (pEntity && pEntity->GetType() == CCLIENTVEHICLE)
     {
         return static_cast<CClientVehicle*>(pEntity);
     }
 
-    return NULL;
+    return nullptr;
 }
 
 CClientVehicle* CClientVehicleManager::GetClosest(CVector& vecPosition, float fRadius)
 {
-    float                                   fClosestDistance = 0.0f;
-    CVector                                 vecVehiclePosition;
-    CClientVehicle*                         pClosest = NULL;
-    vector<CClientVehicle*>::const_iterator iter = m_List.begin();
-    for (; iter != m_List.end(); iter++)
+    float           fClosestDistance = 0.0f;
+    CVector         vecVehiclePosition;
+    CClientVehicle* pClosest = nullptr;
+    for (const auto& iter : m_List)
     {
-        (*iter)->GetPosition(vecVehiclePosition);
-        float fDistance = DistanceBetweenPoints3D(vecPosition, vecVehiclePosition);
+        iter->GetPosition(vecVehiclePosition);
+        const auto fDistance = DistanceBetweenPoints3D(vecPosition, vecVehiclePosition);
         if (fDistance <= fRadius)
         {
-            if (pClosest == NULL || fDistance < fClosestDistance)
+            if (pClosest == nullptr || fDistance < fClosestDistance)
             {
-                pClosest = *iter;
+                pClosest = iter;
                 fClosestDistance = fDistance;
             }
         }
     }
     return pClosest;
 }
 
-bool CClientVehicleManager::IsTrainModel(unsigned long ulModel)
+bool CClientVehicleManager::IsTrainModel(unsigned short usModel)
 {
-    return (ulModel == 449 || ulModel == 537 || ulModel == 538 || ulModel == 569 || ulModel == 590 || ulModel == 570);
+    return (usModel == 449 || usModel == 537 || usModel == 538 || usModel == 569 || usModel == 590 || usModel == 570);
 }
 
-bool CClientVehicleManager::IsValidModel(unsigned long ulModel)
+bool CClientVehicleManager::IsValidModel(unsigned short usModel)
 {
-    CModelInfo* pModelInfo = g_pGame->GetModelInfo(ulModel);
+    const auto pModelInfo = g_pGame->GetModelInfo(usModel);
     return pModelInfo && pModelInfo->IsVehicle();
 }
 
-bool CClientVehicleManager::IsStandardModel(unsigned long ulModel)
+bool CClientVehicleManager::IsStandardModel(unsigned short usModel)
```

Comment:
ushort is a subset of ulong. Are you sure that 65536 models enough?

Reply:
The same above

Reply:
> ushort является подмножеством ulong. Вы уверены, что 65536 моделей достаточно?

Enough

Reply:
In GTA limit 19999